tcms/xmlrpc/tests/test_testrun.py

```diff
@@ -202,3 +203,39 @@ def test_create_with_product_version(self):
         # the result is still using product_version from TR.plan.product_version
         # not the one we specified above
         self.assertEqual(result['product_version'], self.plan.product_version.value)
+
+
+class TestUpdateTestRun(XmlrpcAPIBaseTest):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.initial_test_plan = TestPlanFactory()
+        self.initial_build = BuildFactory()
+        self.initial_summary = 'Initial summary.'
+        self.test_run = TestRunFactory(
+            plan=self.initial_test_plan,
+            build=self.initial_build,
+            summary=self.initial_summary,
+        )
```

Comment:
You don't have to specify initial values for `self.test_run` attributes. When you don't do this the factory will do it for you, and these values will still be different from the ones you are using when performing the update.

Reply:
Further more you are not asserting on these initial values so you don't actually know that the object has been updated. You can say for certain that the latest values in DB match your RPC parameters but not that they are different from what was there before.